Automattic/jetpack
# Admin Menu: Centralize class loading in Jetpack codebase

#### Changes proposed in this Pull Request:
As agreed on paYJgx-1t4-p2, and, in order to simplify the development experience, Jetpack should be the single source of truth for all the Nav Unification logic.

To that end, this PR adds the admin menu class for P2 sites that currently lives in the WP.com codebase. From now on, changes to that menu should be done in Jetpack. cc @Automattic/lighthouse 

This PR also adds the WP.com logic for determining which admin menu class should be instantiated, so we can re-use it in WP.com as well (see D59179-code). Furthermore, this class is filterable through the new `jetpack_admin_menu_class` filter (useful for Atomic hotfixes if we want to override the admin menu behavior).

#### Jetpack product discussion
paYJgx-1t4-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
This PR does not changes the current behavior. Do a smoke test across multiple sites and make sure that there are no regressions.

_Simple_
- Apply D59179-code to your WP.com sandbox.
- Sandbox the API, a domain-only site, a P2 site, and a simple site.
- Go to https://wordpress.com and switch any of the sandboxed sites.
- Check the menu under different scenarios (Calypso, WP Admin, toggle on/off) and make sure there are no regressions.
- Repeat with the rest of the sandboxed sites.

_Atomic_
- Install Jetpack Beta on an Atomic site and switch to the branch of this PR.
- Go to https://wordpress.com and switch to the WoA site.
- Check the menu under different scenarios (Calypso, WP Admin, toggle on/off) and make sure there are no regressions.
